presto-main/src/main/java/com/facebook/presto/sql/planner/LogicalPlanner.java

```diff
@@ -70,6 +70,11 @@
 
 public class LogicalPlanner
 {
+    public enum Stage
+    {
+        CREATED, OPTIMIZED, OPTIMIZED_AND_VALIDATED
```

Comment:
This should probably in a separate commit.

Also, can you remind me of the rationale for this? I remember discussing about this, but I can't find the original comments.